### ISSUE TITLE
[no-unused-vars] False positive: type import used in typeof

### ISSUE THREAD

**Issue by DisappearedStar** [typescript-eslint/typescript-eslint#2453]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

.eslintrc.js
```js
{
  rules: {
    'no-unused-vars': 'off',
    '@typescript-eslint/no-unused-vars': ['error', {
        varsIgnorePattern: '^_',
        argsIgnorePattern: '^_',
        ignoreRestSiblings: true,
    }],
  }
}
```
tsconfig.json
```json
{
  "compilerOptions": {
    "module": "esnext",
    "moduleResolution": "node",
    "target": "es5",
    "allowJs": true,
    "importHelpers": true,
    "jsx": "react",
    "experimentalDecorators": true,
    "isolatedModules": true,
    "importsNotUsedAsValues": "error"
  }
}
```

```TS
// a.ts
export const foo = 42;

// b.ts
import type { foo } from './a';

export type Bar = typeof foo;
```

**Expected Result**

No lint errors

**Actual Result**

`b.ts 1:15 'foo' is defined but never used. Allowed unused vars must match /^_/u. (@typescript-eslint/no-unused-vars)`

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.0.1` |
| `@typescript-eslint/parser`        | `4.0.1` |
| `TypeScript`                       | `4.0.2` |
| `ESLint`                           | `7.8.0` |
| `node`                             | `10.16.3` |

**Comment by bradzacher** [comment]
This is stupidly weird and not all what I expect.
I'm really surprised that it works this way.

When I see `import type { foo }`, I expect that this should either:
1) error, because `foo` is a value (i.e. you can't import a value as a type)
2) be fine but `foo` and under the hood TS would treat it the same as `typeof foo`.

Instead for all intents and purposes, TS treats `foo` as a value, however it only allows `foo` be referenced in type contexts.

This breaks the assumption that I'd made that all `import type`s are 100% types.
No biggie - easy fix, it just feels wrong.

**Comment by DisappearedStar** [comment]
I confirm that the error is fixed in 4.0.2-alpha.9.

**Comment by Wolk** [comment]
Hi, i am still having this bug in 4.0.3

example of my problem
```ts
// file 1
export type FieldDefinition = {
  [key: string]: string[];
};

// file 2
interface FruitsFormDefinition {
  apple: string;
  orange: string;
}

export type FruitsFieldDefinition = { [K in keyof FruitsFormDefinition]: FieldDefinition[string] };

export const fruitsFieldDefiniton: FruitsFieldDefinition = {
  apple: [],
  orange: ['isRequired'],
};
```

the error i get is 
  Line 32:37:  'K' is defined but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars

**Comment by OriginLive** [comment]
^ made a new ticket, since it's not directly related https://github.com/typescript-eslint/typescript-eslint/issues/2615
